### PR TITLE
feat(selector): expose integration API for homepage showcase

### LINKED ISSUE #680: feat(selector): expose integration API for homepage showcase
## Summary

Give the showcase page a supported way to drive theming through the theme-selector package instead of the prototype's parallel path (inline `define:vars` script poking dropdown internals + `window.__showcaseMeta` global).

## Problem

The WIP baseline duplicates theme-switching responsibilities: icon/label updates, `data-theme`/`data-appearance` handling, and metadata distribution all happen twice. Duplication will drift as themes evolve.

## Proposed Solution

- Public API (or documented custom events) on theme-selector for: apply theme by id, query current theme, subscribe to theme changes
- Showcase marquee clicks call this API; theme-selector owns persistence and dropdown state
- Delete `window.__showcaseMeta` and the duplicated icon/label logic from the page's inline script

## Implementation Notes

- `packages/theme-selector` — keep site-only UI out of the published package; only the API surface belongs here
- Depends on nothing; #683 consumes it

## Benefits

- One theming source of truth
- Showcase survives theme additions/removals without edits

### PR DESCRIPTION
**Pull request by TurboCoder13**
## Summary

Adds a small public integration API to `packages/theme-selector` so the upcoming homepage showcase (#683) can drive theming through the package instead of the prototype's parallel path (inline script poking dropdown internals + `window.__showcaseMeta` global).

New exports from `@lgtm-hq/turbo-theme-selector`:

- `applyTheme(themeId, documentObj?, windowObj?): Promise<boolean>` — validates the ID (format + registry), persists the selection, applies `data-theme`/`data-appearance` and theme CSS, syncs dropdown state, and notifies subscribers. Returns `false` (no side effects, no event) for unknown or malformed IDs.
- `getCurrentTheme(documentObj?): string` — reads the active theme from the `data-theme` root attribute, falling back to the `theme-*` class, then the default theme.
- `subscribeToThemeChanges(listener, documentObj?): Unsubscribe` — explicit subscribe backed by a `turbo-theme-change` CustomEvent on `document`; the listener receives `{ themeId, appearance }`.
- `THEME_CHANGE_EVENT` constant plus `ThemeChangeDetail`, `ThemeChangeListener`, and `Unsubscribe` types.

Dropdown selections now route through the same `applyTheme` path, so subscribers are notified regardless of whether the change came from the API or the selector UI. The package remains the sole owner of persistence, root attributes, and dropdown state; no site-only UI was added to the package, and no live site pages were touched.

## Testing

- New `packages/theme-selector/test/integration.test.ts` with 23 tests covering apply, query, subscribe/unsubscribe, and invalid theme ID handling (selector suite: 340 → 363 tests)
- `uv run lintro chk`: 0 issues
- `bun run build`: passes
- `bun run test`: 94 files, 3175 tests, all passing

Closes #680

<details><summary><h3>Greptile Summary</h3></summary>

This PR adds a clean public integration API to `packages/theme-selector` — `applyTheme`, `getCurrentTheme`, `subscribeToThemeChanges`, `THEME_CHANGE_EVENT`, and associated types — so the upcoming homepage showcase can drive theming without coupling to internal selector mechanisms. Dropdown selections now route through the same `applyTheme` path, giving subscribers consistent notifications regardless of the change origin.

- **New `integration.ts`** exports `applyTheme` (validates ID, persists via `saveTheme`, applies DOM state, emits `turbo-theme-change`), `getCurrentTheme` (reads `data-theme` attribute then theme class, falls back to default), and `subscribeToThemeChanges` (wraps a `CustomEvent` listener with a guard against bare `Event` dispatches).
- **`index.ts` updated** to wire `wireFlavorSelector`'s `onThemeSelect` through the new integration `applyTheme` and re-export all public API symbols; `initTheme` deliberately retains the direct internal path and does not emit `turbo-theme-change` on initial load.
- **23 new integration tests** cover valid/invalid apply, DOM attribute assertions, localStorage persistence, event dispatch, multi-subscriber semantics, and edge cases including markup-injection IDs and post-unsubscribe silence.
</details>

<h3>Confidence Score: 4/5</h3>

Safe to merge with awareness of the two open issues in integration.ts noted in prior review threads.

Two open issues from prior threads remain unresolved in integration.ts: CSS load errors are swallowed and the resulting DOM-attribute change is silently reported as a successful application, and the saveTheme boolean return is discarded, meaning a failed localStorage write still yields true and fires the change event. Both affect the reliability of the API contract for the showcase consumer. Outside of those, the integration is well-structured, the new test suite is thorough, and the bundle output matches the TypeScript source.

packages/theme-selector/src/integration.ts — the unresolved CSS load-failure and persistence-failure handling in applyTheme.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/theme-selector/src/integration.ts | New public integration API; two unresolved issues flagged in prior review threads: CSS load failures are swallowed and reported as success, and saveTheme's boolean return is silently discarded. Core logic is otherwise sound. |
| packages/theme-selector/src/index.ts | Wires dropdown onThemeSelect through the new integration applyTheme path and re-exports all public API symbols; initTheme still uses the direct internal path and does not emit turbo-theme-change. |
| packages/theme-selector/test/integration.test.ts | 23 tests covering valid/invalid apply, DOM attribute assertions, localStorage persistence, event dispatch, subscribe/unsubscribe, and edge cases; no coverage for CSS load failure or saveTheme unavailability. |
| assets/js/theme-selector.js | Bundled output matches the TypeScript source for all new exports and internal renames; no discrepancies between the source and built artifact. |
| assets/js/theme-selector.min.js | Minified build artifact corresponding to the source changes; no independent issues. |
| assets/js/theme-selector.js.map | Source map for the updated bundle; machine-generated, no review concerns. |

</details>

<details><summary><h3>Sequence Diagram</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
sequenceDiagram
    participant Caller as Showcase
    participant API as integration.ts
    participant Resolver as theme-resolver
    participant Storage as storage
    participant Apply as apply-theme
    participant Loader as theme-loader
    participant Sub as subscriber

    Caller->>API: applyTheme(themeId)
    API->>Resolver: isValidThemeId + getValidThemeIds
    alt invalid or unknown ID
        API-->>Caller: return false
    else valid ID
        API->>Storage: saveTheme(win, themeId, validIds)
        API->>Apply: await applyThemeToDocument(doc, themeId)
        Apply->>Loader: applyThemeClass sets data-theme immediately
        Apply->>Loader: await loadThemeCSS
        alt CSS load fails
            Loader-->>Apply: error logged and swallowed
        end
        Apply-->>API: resolves void
        API->>Sub: emitThemeChange dispatches turbo-theme-change
        Sub-->>Caller: listener receives themeId and appearance
        API-->>Caller: return true
    end
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
sequenceDiagram
    participant Caller as Showcase
    participant API as integration.ts
    participant Resolver as theme-resolver
    participant Storage as storage
    participant Apply as apply-theme
    participant Loader as theme-loader
    participant Sub as subscriber

    Caller->>API: applyTheme(themeId)
    API->>Resolver: isValidThemeId + getValidThemeIds
    alt invalid or unknown ID
        API-->>Caller: return false
    else valid ID
        API->>Storage: saveTheme(win, themeId, validIds)
        API->>Apply: await applyThemeToDocument(doc, themeId)
        Apply->>Loader: applyThemeClass sets data-theme immediately
        Apply->>Loader: await loadThemeCSS
        alt CSS load fails
            Loader-->>Apply: error logged and swallowed
        end
        Apply-->>API: resolves void
        API->>Sub: emitThemeChange dispatches turbo-theme-change
        Sub-->>Caller: listener receives themeId and appearance
        API-->>Caller: return true
    end
```

</a>
</details>

<sub>Reviews (2): Last reviewed commit: ["build(selector): regenerate theme-select..."](https://github.com/lgtm-hq/turbo-themes/commit/e69774da72c86f2655b48db8938ceab973791435) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=45615610)</sub>

### CODE REVIEW

File: packages/theme-selector/src/integration.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+ * @param themeId - ID of the theme to apply (e.g. `catppuccin-mocha`)
+ * @param documentObj - Document to apply the theme to (defaults to global)
+ * @param windowObj - Window used for persistence (defaults to global)
+ * @returns true if the theme was applied, false for invalid theme IDs
+ */
+export async function applyTheme(
+  themeId: string,
+  documentObj: Document = document,
+  windowObj: Window = window,
+): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply packages/theme-selector/src/integration.ts:72]: Agreed the return value overstates success. The internal swallow in loadThemeCSS is pre-existing package behavior, so changing propagation semantics deserves its own scoped change rather than a rider on this API PR — tracked in #695 (richer result or false on partial failure, with tests).

File: packages/theme-selector/src/integration.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  const validIds = getValidThemeIds();
+  if (!isValidThemeId(themeId) || !validIds.has(themeId)) {
+    logThemeError(ThemeErrors.INVALID_THEME_ID(themeId));
+    return false;
+  }
+
+  saveTheme(windowObj, themeId, validIds);
+  await applyThemeToDocument(documentObj, themeId);
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply packages/theme-selector/src/integration.ts:80]: Valid — saveTheme's false (private browsing/quota) is ignored and the docstring claims persistence. Folded into #695 together with the CSS-load case so the API reports partial failure consistently.
